From 076a97aa32492cc44e863f7ab75494dc0b3bf5ef Mon Sep 17 00:00:00 2001
From: Guojing Yuan <guojing@google.com>
Date: Tue, 22 Oct 2024 17:06:48 +0000
Subject: [PATCH] [CDM] Check if it's system UID for hidden Perm Sync APIs

Bug: 371975420
Test: Manual test
Flag: EXEMPT bugfix
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:08f57da9e39869f696de4225197530aa2b9187e8)
Merged-In: I9771ce38728d40b026d94cb6785d55f144c66790
Change-Id: I9771ce38728d40b026d94cb6785d55f144c66790
---
 .../server/companion/CompanionDeviceManagerService.java  | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/services/companion/java/com/android/server/companion/CompanionDeviceManagerService.java b/services/companion/java/com/android/server/companion/CompanionDeviceManagerService.java
index 0ab6bbc3e0d3..57eadf65f2a0 100644
--- a/services/companion/java/com/android/server/companion/CompanionDeviceManagerService.java
+++ b/services/companion/java/com/android/server/companion/CompanionDeviceManagerService.java
@@ -628,16 +628,25 @@ public class CompanionDeviceManagerService extends SystemService {
 
         @Override
         public void enablePermissionsSync(int associationId) {
+            if (getCallingUid() != SYSTEM_UID) {
+                throw new SecurityException("Caller must be system UID");
+            }
             mSystemDataTransferProcessor.enablePermissionsSync(associationId);
         }
 
         @Override
         public void disablePermissionsSync(int associationId) {
+            if (getCallingUid() != SYSTEM_UID) {
+                throw new SecurityException("Caller must be system UID");
+            }
             mSystemDataTransferProcessor.disablePermissionsSync(associationId);
         }
 
         @Override
         public PermissionSyncRequest getPermissionSyncRequest(int associationId) {
+            if (getCallingUid() != SYSTEM_UID) {
+                throw new SecurityException("Caller must be system UID");
+            }
             return mSystemDataTransferProcessor.getPermissionSyncRequest(associationId);
         }
 
-- 
2.46.1.824.gd892dcdcdd-goog

